### PR TITLE
Feat/domain message flow

### PR DESCRIPTION
**Pull request by neolytian**
Add our take of the Domain-Message Flow

### CODE REVIEW

File: detailedinfo/domainmessageflow.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+If you find any issues or areas for improvement, you can revise your diagrams and update your message flows accordingly. You can also get feedback from other developers, architects, or domain experts to validate and refine your message flows. By doing this, you can ensure that your message flows are accurate, consistent, and comprehensive. With this iterative approach you can also verify if your EventStorming has been valid. You can take the newly gained insights to iterate on your EventStorming.
+
+## Anti-Pattern
+
+An anti-pattern is a common but ineffective or harmful solution to a problem, that has more negative consequences than positive ones. It is the opposite of a pattern, which is a good and proven solution to a problem. Anti-patterns can be found in various domains, such as software engineering, project management, and business processes. The following are anti-pattern we have identified and observed in several workshops. They are often indicators that the domain is not broken down into independent bounded contexts. By not addressing them you might inflict unwanted and unneeded dependencies into your application design.
+
+### Synchronous Dependencies / Communication
```
Comment (mr-flannery) [inline detailedinfo/domainmessageflow.md:50]: I think this paragraph overshoots the mark a little and needs a bit more nuance. 

First, I think it's important to differentiate between synchronous requests at a technical and at a process level. I do agree that when a process has to wait for a synchronous response from some other service in order to do it's job and the unavailability of said component leads to an error that is observable by clients, that is a problem and can be solved by designing the process asynchronously.

However, such design can be achieved with synchronous means like HTTP feeds or transactional outbox. So saying "synchronous = bad" is too coarse and I think can lead to wrong conclusions like "okay I need Kafka/rabbitmq/whatever everywhere".

File: detailedinfo/domainmessageflow.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+Synchronous communication pattern is a common but problematic way of designing interactions between bounded contexts, which are sub-systems that represent a specific part of the domain. Synchronous communication pattern means that the sender and the receiver of a message must be online and ready at the same time, and that the sender must wait for a reply before continuing with other tasks. This pattern can have several negative effects on the system, such as:
```
Comment (mr-flannery) [inline detailedinfo/domainmessageflow.md:52]:
```suggestion
The synchronous communication pattern is a common but problematic way of designing interactions between bounded contexts, which are sub-systems that represent a specific part of the domain. Synchronous communication pattern means that the sender and the receiver of a message must be online and ready at the same time, and that the sender must wait for a reply before continuing with other tasks. This pattern can have several negative effects on the system, such as:
```

File: detailedinfo/domainmessageflow.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+- It can reduce the performance and the scalability of the system, as each message can block the execution of other tasks or processes until a response is received, which can cause delays, bottlenecks, and congestion in the system.
+- It can affect the reliability and the resilience of the system, as each message can be vulnerable to errors, failures, or disruptions in the communication channel or the message processing, which can cause the system to fail or behave unpredictably.
+- It can limit the flexibility and the adaptability of the system, as each message can impose strict constraints on the availability, compatibility, and coordination of the communicating parties, which can prevent the system from evolving or changing according to the needs of the domain.
+
+Therefore, synchronous communication pattern is an anti-pattern that should be avoided or replaced by a better solution, such as asynchronous communication pattern, which is more suitable for loosely-coupled systems.
```
Comment (mr-flannery) [inline detailedinfo/domainmessageflow.md:60]:
```suggestion
Therefore, the synchronous communication pattern is an anti-pattern that should be avoided or replaced by a better solution, such as the asynchronous communication pattern, which is more suitable for loosely-coupled systems.
```

**Review by mr-flannery — APPROVED** [review]
Mostly very good 👍 Some suggestions for typos/phrasing, some content related questions/concerns

File: detailedinfo/domainmessageflow.md (hunk 1 above)
Comment (neolytian) [reply detailedinfo/domainmessageflow.md:50]: Changed it

**Pull request merged**